spec/openapi-spec.yaml

```diff
@@ -0,0 +1,40 @@
+openapi: 3.1.1
+info:
+  title: Lance REST Catalog API
+  version: 0.0.1
+  description: REST interface for managing Lance dataset or LanceDB databases and tables
+
+paths:
+  /v1/databases:
+    post:
+      tags: [ Database ]
+      summary: Create new database
+      operationId: CreateDatabase
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateDatabaseRequest'
+      responses:
+        201:
```

Comment:
why not return 200 with a response, vs returning 201 with a header for the info?

Reply:
Yes, 200 is also OK, but it seems 201 looks more suitable for the RESTful spec. It means we create a resource.

FYI: https://restful.p2hp.com/resources/http-status-codes/http-status-201-created

However, I know it's a trade-off. But not sure if it's not a good choice in the future.

What's your opinion?

Reply:
I thought about that a bit more, I think it might make more sense to return the object, for 2 reasons:
1. additional system metadata: it is common to add additional info at server side, like `created_by`, `created_at`, `last_updated_by`, `last_updated_at`. These information can be returned as the result of a creation, rather than having to do another GetDatabase to retrieve the information.
2. consistency with CreateTable: CreateTable requires returning the object created, because when used in CREATE TABLE AS SELECT (or more complicated transactions), the API is responsible to perform a "staging create", and the user is supposed to use the response of CreateTable to perform subsequent actions and eventually commit the complete content.

Reply:
sounds reasonable